cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/GenericStepDefinitions.java

```diff
@@ -1,19 +1,88 @@
 package com.skillstorm.StepDefinitions;
 
+import io.cucumber.java.en.*;
 
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.Select;
+import org.openqa.selenium.support.ui.WebDriverWait;
 
-public class GenericStepDefinitions {
+import com.skillstorm.Utilities.Navigator;
 
-    // @Given("an example scenario")
-    // public void anExampleScenario() {
-    // }
 
-    // @When("all step definitions are implemented")
-    // public void allStepDefinitionsAreImplemented() {
-    // }
+//This class implements some step definitions that are commonly used among our tests. You must pass in the web driver you're using as a parameter.
+public class GenericStepDefinitions{
 
-    // @Then("the scenario passes")
-    // public void theScenarioPasses() {
-    // }
+    WebDriver driver;
+    WebDriverWait wait;
+    Navigator navigator = new Navigator(driver);
 
+    public GenericStepDefinitions(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    /*
+     * General use definition for page loading step. Uses navigator class in Utilities package
+     * -------------------------------------------------------------------------------------------------------------
+     * Parameters:
+     * pageName - the page name you'd want to navigate to. It must match Navigator.java naming convention.
+     * -------------------------------------------------------------------------------------------------------------
+     * If you haven't already, you must redefine the method for navigation for your respective page and set the URL for your page as well.
+     */
+
+    @Given("I am on the {string} page")
+    public void iAmOnThePage(String pageName){
+        navigator.navigateTo(pageName);
+    }
+
+
+    /*
+     * General use definition for button click step.
+     * -------------------------------------------------------------------------------------------------------------
+     * Parameters:
+     * buttonId - pass in the buttonId for the button you'd like to press.
+     * -------------------------------------------------------------------------------------------------------------
+     */
+    
+    
+    @When("I click the {string} button")
+    public void iClickTheButton(String buttonId){
+        WebElement button = driver.findElement(By.id(buttonId));
+        button.click();
+    }
```

Comment:
Since each persons implementation of clicking a button name should be different (we don't know if we will be able to id ALL the buttons with specific names, should this implementation be something like:
public void iClickTheButton(String buttonName){
    page.clickButton(buttonName);
}

//The "page" object (which are the POM objects we are creating) could be loaded in people's "@Given" steps for example